cmake/ConfigureBenchmarks.cmake

```diff
@@ -7,4 +7,36 @@ function(configure_benchmark_executable EXECUTABLE_NAME)
     target_include_directories(${EXECUTABLE_NAME} PRIVATE ${PROJECT_INCLUDE_FOLDER})
     target_link_libraries(${EXECUTABLE_NAME} PRIVATE ${LIBRARY_NAME})
 
+    # Link google benchmarks
+    link_libraries("-lbenchmark")
```

Comment:
Shouldn't this be `target_link_libraries(${EXECUTABLE_NAME} "-lbenchmark")`?

Reply:
Furthermore, can we do `find_package(google_benchmark)`?

Reply:
yes, "find_package(benchmark)" should we "REQUIRE" it? as you don't actually need it (only for benchmarks)

Reply:
No, I don't think you should `REQUIRE` it, I think you specify it as `OPTIONAL`.